api/template.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+import "time"
+
+// GuildTemplate is a template used for copying guilds https://discord.com/developers/docs/resources/guild-template
+type GuildTemplate struct {
+	Code          string    `json:"code"`
+	Name          string    `json:"name"`
+	Description   *string   `json:"description,omitempty"`
+	UsageCount    int       `json:"usage_count"`
+	CreatorID     Snowflake `json:"creator_id"`
+	Creator       User      `json:"creator"`
```

Comment:
tbh this should be thrown into the EntityBuilder
```suggestion
	Creator       *User     `json:"creator"`
```